src/rosdiscover/recover/loader.py

```diff
@@ -220,9 +222,24 @@ def _load_statement(self, dict_: t.Mapping[str, t.Any]) -> SymbolicStatement:
             return self._load_checks_for_param(dict_)
         elif kind == "compound":
             return self._load_compound(dict_)
+        elif kind == "while":
+            return self._load_while(dict_)
+        elif kind == "if":
+            return self._load_if(dict_)
         else:
             raise ValueError(f"unknown statement kind: {kind}")
 
+    def _load_while(self, dict_: t.Mapping[str, t.Any]) -> SymbolicWhile:
+        assert dict_["kind"] == "while"
+        return SymbolicWhile(self._load_compound(dict_["body"]), self._load_value(dict_["condition"]))
+
+    def _load_if(self, dict_: t.Mapping[str, t.Any]) -> SymbolicIf:
+        assert dict_["kind"] == "if"
+        return SymbolicIf(
+            true_body=self._load_compound(dict_["trueBranchBody"]),
+            false_body=self._load_compound(dict_["falseBranchBody"]),
+            condition=self._load_value(dict_["condition"]))
```

Comment:
nitpick for style consistency: add a trailing comma to the condition line and put the closing bracket on its own line below

Reply:
fixed